src/java-opentimelineio/src/main/java/io/opentimeline/opentimelineio/AnyDictionary.java

```diff
@@ -0,0 +1,279 @@
+    public Any get(Object o) {
+        if (o instanceof String)
+            return get((String) o);
+        return null;
+    }
+
+    public native int size();
+
+    public native void clear();
+
+    @Override
+    public Set<String> keySet() {
+        Set<String> keys = new HashSet<>();
+        Iterator iterator = iterator();
+        while (iterator.hasNext()) {
+            AnyEntry element = iterator.next();
+            keys.add(element.getKey());
+        }
+        try {
+            iterator.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return keys;
+    }
+
+    @Override
+    public Collection<Any> values() {
+        AnyVector anyVector = new AnyVector();
+        Iterator iterator = iterator();
+        while (iterator.hasNext()) {
+            AnyEntry element = iterator.next();
+            anyVector.add(element.getValue());
+        }
+        try {
+            iterator.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return anyVector;
+    }
+
+    @Override
+    public Set<Entry<String, Any>> entrySet() {
+        Set<Entry<String, Any>> elements = new HashSet<>();
+        Iterator iterator = iterator();
+        while (iterator.hasNext()) {
+            elements.add(iterator.next());
+        }
+        try {
+            iterator.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return elements;
+    }
+
+    @Override
+    public void forEach(BiConsumer<? super String, ? super Any> action) {
+        Iterator iterator = iterator();
+        while (iterator.hasNext()) {
+            AnyEntry anyEntry = iterator.next();
+            action.accept(anyEntry.getKey(), anyEntry.getValue());
+        }
+        try {
+            iterator.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public native int remove(String key);
+
+    public boolean equals(AnyDictionary anyDictionary) {
+        if (size() != anyDictionary.size()) return false;
+
+        Iterator thisIterator = iterator();
+        Iterator otherIterator = anyDictionary.iterator();
+
+        while (thisIterator.hasNext() && otherIterator.hasNext()) {
+            AnyEntry thisElement = thisIterator.next();
+            AnyEntry otherElement = otherIterator.next();
+            if (!(thisElement.key.equals(otherElement.key))
+                    || !thisElement.value.equals(otherElement.value))
+                return false;
+        }
+        try {
+            thisIterator.close();
+            otherIterator.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof AnyDictionary))
+            return false;
+        return this.equals((AnyDictionary) obj);
+    }
+
+    @Override
+    public String toString() {
```

Comment:
for more conventional java way you can use lambda streams with collector:
```
 StreamSupport.stream(iterable.spliterator(), false)
                .map(thisElement => thisElement.getKey()).append("=").append(thisElement.getValue())
                Collectors.joining( ", " );
```

Reply:
Thanks! This is great!